satpy/dataset/metadata.py

```diff
@@ -141,6 +145,29 @@ def _all_arrays_equal(arrays):
         return _pairwise_all(nan_allclose, arrays)
 
 
+def _all_dicts_equal(dicts):
+    return _pairwise_all(_dict_equal, dicts)
+
+
+def _dict_equal(d1, d2):
+    """Check that two dictionaries are equal.
+
+    Nested dictionaries are walked recursively. Array type values are compared
+    approximately, other types are checked for identity.
+    """
+    if not (isinstance(d1, dict) and isinstance(d2, dict)):
+        return False
+    if not d1.keys() == d2.keys():
+        return False
+    for key in d1.keys():
+        if isinstance(d1[key], dict) and isinstance(d2[key], dict):
```

Comment:
Could we reuse `_are_values_combinable` in some way to replace this logic in the for loop? That function doesn't actually do the assignment so maybe another one has to be created to help with that. I admit that the misdirection/hand-wavy nature of the recursion does get really bad by doing that though. Just a thought.